pkg/network/multus/status.go

```diff
@@ -29,6 +29,17 @@ import (
 	"kubevirt.io/client-go/log"
 )
 
+// NetworkStatusesByPodIfaceName creates a map of NetworkStatus objects by pod interface name.
+func NetworkStatusesByPodIfaceName(networkStatuses []networkv1.NetworkStatus) map[string]networkv1.NetworkStatus {
```

Comment:
According to [Kubernetes Network Custom Resource Definition De-facto Standard](https://github.com/k8snetworkplumbingwg/multi-net-spec/blob/master/v1.3/%5Bv1.3%5D%20Kubernetes%20Network%20Custom%20Resource%20Definition%20De-facto%20Standard.pdf):

> 4. Network Attachment Selection Annotation
To select one or more secondary ("sidecar") networks to which a pod should be attached, this specification defines a Pod object annotation

When using secondary networks (or binding plugins with CNI) in KubeVirt, we use the `k8s.v1.cni.cncf.io/networks` annotation to connect to these networks.

The value of `k8s.v1.cni.cncf.io/network-status` represents the successful network attachments - meaning that multiple network attachments could be represented by it.
This is the reason it unmarshals to a `[]networkv1.NetworkStatus` - one element per network.

One could look at the entire slice and consider it as the overall "pod network status", but a different approach could consider each element as a single network status.

Reply:
Its not about approach, semantically this slice represent a pod network status, the fact that it has multiple interfaces, the slice as whole represent the pod network status.
In addition network-stautses reads bad, I think the term should be network-status because in all contexts we refer to this slice as the pod network status (as whole).

If you want to stick with referring to the slice as each interface status, I think the correct form is network**s**-status.

Reply:
> Its not about approach I think the definition of the network-status slice is interpreted wrongly here.
Semantically this slice represent a pod network status, the fact that it has multiple interfaces, the slice as whole represent the pod network status.

How would you refer to each entry in case the whole slice is called `networkStatus`?

> In addition network-stautses reads bad, I think the term should be network-status because in all contexts we refer to this slice as the pod network status (as whole).

Following #13018 and this PR - it is referred to it as `networkStatuses` all over.

Reply:
> How would you refer to each entry in case the whole slice is called networkStatus?

If you refer the whole slice as `networksStatus`, each element could be referred to `networkIfaceStatus`.
I also suggested to refer to the slice as `networksStatus` as an alternative, I think its the correct form (having the last adjective plural).

> Following https://github.com/kubevirt/kubevirt/pull/13018 and this PR - it is referred to as `networkStatuses` all over.

I am aware, I also commented in previous PRs, I am raising it again because it getting to other places.
Hopefully I will manage to convince you this time :slightly_smiling_face:

Reply:
I do not understand the amount of investment in this thread.

You can convince me by ref the multus-cni source code or by being focused (avoiding lengthy explanations about the meaning).

My only request is to wrap this up quickly and try to prefer short over long names when possible:

net over network
iface over interface

Reply:
Long story short I think the `networkStatuses` is wrong, correct form is `networksStatus`, please consider changing it instead of adding more of the former naming.

Reply:
From the Multus source code:
https://github.com/k8snetworkplumbingwg/multus-cni/blob/ecf5854ca9ce4c064ab4939808868747d7fa438f/vendor/github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/utils/net-attach-def.go#L49-L48

I've considered, and decided `networkStatuses` makes sense.
If there aren't additional issues with the PR, I suggest to move forward with it.
A follow-up PR could suggest renaming all the instances of this slice.

Reply:
I am still think its not the right form but not blocking this PR on it

Reply:
Thank you @ormergi.